basicLibrary/src/main/java/basicLibrary/Library.java

```diff
@@ -84,12 +84,43 @@ public static String analyzeWeatherData(int[][] weeklyMonthTemperatures) {
         String message = String.format("High: %d %nLow: %d", highestTemp, lowestTemp);
 
         // Now to reveal which temps are not in the data set
-
         for (int i = lowestTemp; i < highestTemp; i++) {
             if (!temps.contains(i)) {
                 message += String.format("%nNever saw temperature: %d", i);
             }
         }
         return message;
     }
+
+    public static String TallyVotes(List<String> votes){
+        String winner = "";
+        int winnerVotes = 0;
+
+        HashMap<String, Integer> candidates = new HashMap<>();
+
+        // tally votes, add to hash map
+        for (String vote : votes) {
+            if (candidates.containsKey(vote)) {
+                int temp = candidates.get(vote);
```

Comment:
When you can, it's a good idea to create variable outside of loops. That way, instead of creating and assigning a values each time it loops, it's just assigning values (this is kind of a micro-efficiency booster, in the grand scheme of things it won't effect much)